aten/src/ATen/native/native_functions.yaml

```diff
@@ -2154,6 +2154,10 @@
 
 - func: eye.m(int n, int m, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
 
+- func: eye.nk(int n, *, int? k, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
```

Comment:
in this signature `k` isn't optinal but required as a kwarg, right?

Reply:
Yup, `k` isn't actually optional. Unfortunately:

```c++
 Tensor& eye_out_cuda(int64_t n, int64_t m, Tensor& result) {                                                                                                                           
   // the default value of `k` equals to 0                                                                                                                                              
   return at::native::eye_out_cuda(n, m, 0, result);                                                                                                                                    
 }                                                                                                                                                                                      
                                                                                                                                                                                        
Tensor& eye_out_cuda_nk(int64_t n, int64_t k, Tensor& result) {                                                                                                                        
   // the default value of `m` equals to `n`                                                                                                                                            
   int64_t k_v{k.has_value() ? k.value() : 0};                                                                                                                                          
   return at::native::eye_out_cuda(n, n, k_v, result);                                                                                                                                  
 }                                                                                                                                                                                      
         
```

These are ambiguous, since there's no way to know if the user passed in `k` or `m`. Actually the signature is:

```
eye(n_rows: int, n_cols: Optional[int] = None, /, *, k: int = 0, dtype: Optional[dtype] = None, device: Optional[device] = None) → array
```

Actually, the Array-API requires that `m` is to be optional, and `k` should be a keyword only argument. For BC reasons I ended up swapping the two; making `k` optional.

Reply:
@IvanYashchuk does this seem like a reasonable approach?

Reply:
There is an ambiguity with the "eye.m" overload at the C++ level as there's no concept of a keyword-only argument. Is it possible to add `eye.overload_name(int n, int? m=None, *, int k, ...)`?